### DISCUSSION TITLE
How to add tags/label for billing purposes and segregate users access with HPC Slurm?

### DISCUSSION THREAD

**Discussion by edl4006** [GoogleCloudPlatform/cluster-toolkit#3020] [General]
I'm looking to test and use HPC Slurm through the cluster-toolkit but have some confusion with this particular things, hoping these are easy questions to be answered.

1. When I spin up more instances from my Slurm , how can I add a unique tag so I can segregate cost and billing associated to the nodes that was just spun up in the GCP Billing console through Slurm?
2. When I spin up instances via Slurm I want users to be able to access their own pool of nodes, is there a way to set up a specific service accounts access when I provision via Slurm or ensure that I can segregate users access somehow based off of slurm provisioning?

Thank you.

**Comment by tpdownes** [comment]
Hi @edl4006 

For billing purposes we advise adding labels to your Slurm nodeset (the schedmd-slurm-gcp-v6-nodeset module) with the following input variable:

https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/community/modules/compute/schedmd-slurm-gcp-v6-nodeset/README.md#input_labels

Each nodeset is automatically labeled with `ghpc_deployment: deployment_name`, where the `deployment_name` matches the value in your blueprint.

If I understand your 2nd question, you can assign custom service accounts to each nodeset following our "enterprise" example":

https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/f9f92565a7a4133e0c7b31cd55e56cb0728dbb2c/examples/hpc-enterprise-slurm.yaml#L74-L111

We do not have any examples of access control on Slurm partitions. Is your goal to create groups of POSIX users who can access Partition A with Service Account A and Partition B with Service Account B?
